omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/resource/IntegrityResultPatientProgramResource.java

```diff
@@ -0,0 +1,118 @@
+    public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+        if (representation instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("actionUrl");
+            description.addProperty("notes");
+            description.addProperty("patient");
+            description.addProperty("patientProgram");
+            description.addProperty("rule");
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (representation instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("actionUrl");
+            description.addProperty("rule");
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (representation instanceof RefRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("actionUrl");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    private DataIntegrityService getDataIntegrityService() {
+        return Context.getService(DataIntegrityService.class);
+    }
+
+    @Override
+    public DataIntegrityResult getParent(DataIntegrityResult dataIntegrityResult) {
+        return dataIntegrityResult;
+    }
+
+    @Override
+    public void setParent(DataIntegrityResult dataIntegrityResult, DataIntegrityResult dataIntegrityResult2) {
+        dataIntegrityResult.setActionUrl(dataIntegrityResult2.getActionUrl());
+        dataIntegrityResult.setDateCreated(dataIntegrityResult2.getDateCreated());
+        dataIntegrityResult.setId(dataIntegrityResult2.getId());
+        dataIntegrityResult.setNotes(dataIntegrityResult2.getNotes());
+        dataIntegrityResult.setPatient(dataIntegrityResult2.getPatient());
+        dataIntegrityResult.setPatientProgram(dataIntegrityResult2.getPatientProgram());
+        dataIntegrityResult.setRule(dataIntegrityResult2.getRule());
+        dataIntegrityResult.setUuid(dataIntegrityResult.getUuid());
+    }
+
+    @Override
+    public PageableResult doGetAll(DataIntegrityResult dataIntegrityResult, RequestContext requestContext) throws ResponseException {
+        String patientUuid = requestContext.getParameter("patientUuid");
+        return new NeedsPaging<>(getDataIntegrityService().getResultsByPatientUuid(patientUuid), requestContext);
```

Comment:
Make sure the response from the `getResultsByPatientUuid()` is listable, otherwise you should create the list before paging.